bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -369,10 +369,9 @@ public Optional<Thing> getLegacyThing(String targetIdV1) {
         return thingRegistry.getAll().stream() //
                 .filter(thing -> legacyBridgeUID.equals(thing.getBridgeUID())
                         && V1_THING_TYPE_UIDS.contains(thing.getThingTypeUID())) //
-                .filter(thing -> {
-                    Object id = thing.getConfiguration().get(config);
-                    return (id instanceof String) && targetIdV1.endsWith("/" + (String) id);
-                }).findFirst();
+                .filter(thing -> thing.getConfiguration().get(config) instanceof String id
+                        && targetIdV1.endsWith("/" + id))
+                .findFirst();
```

Comment:
Unrelated, but is there any reason to have two filters?
```java
        return thingRegistry.getAll().stream() //
                .filter(thing -> legacyBridgeUID.equals(thing.getBridgeUID())
                        && V1_THING_TYPE_UIDS.contains(thing.getThingTypeUID())
                        && thing.getConfiguration().get(config) instanceof String id && targetIdV1.endsWith("/" + id))
                .findFirst();
```

Reply:
Good point. I will refactor it tomorrow.